### PR TITLE
Some config changes

### PR DESCRIPTION
**Pull request by kotarou3**

### CODE REVIEW

**Comment by kotarou3** [comment]
The "Minor refactor to promote code" would be in a separate pull req if not it conflicting with the initial permission alphabetisation, and so you only would have to change the config once.

Also considering camelCasing the config options as well, though you didn't want it last time I asked, so not added to the pull request for now.
There's also my alternate groups structure (https://github.com/kotarou3/Pokemon-Showdown/blob/config-refactor/config/config-example.js#L267), though you didn't like it either last time I asked.

If you've changed your mind for either of those two, I can add it before you accept the pull request so you only have to change your config once.

**Comment by Zarel** [comment]
I think the `promote` permission is kind of too complicated. My system of doing `roomvoice` etc seemed easier to understand.

**Comment by Zarel** [comment]
Is `'u'` documented anywhere?

**Comment by kotarou3** [comment]
Yes, in the `jurisdiction` documentation.

Why do you think the `promote` permission is too complicated? As stated in the documentation, to promote someone, they simply need to have promote jurisdiction over both the source and target groups. That's all there is to it. It also makes the actual promote code a fair bit simpler too

**Comment by kotarou3** [comment]
bump

**Comment by Zarel** [comment]
todo: investigate these changes in more detail

**Comment by kotarou3** [comment]
Also: https://github.com/Zarel/Pokemon-Showdown/commit/ad0325e736b8a838f71bce6e663bfd8c3fb190d7#commitcomment-16941289

**Comment by kotarou3** [comment]
bump

**Comment by Lord-Haji** [comment]
waoh feelscool
u need to resolve the commits tho

**Comment by Zarel** [comment]
These are actually really good changes but unfortunately I left this on the back-burner for too long...

**Comment by Zarel** [comment]
Also this doesn't appear to be backwards-compatible...

**Comment by DeathlyPlays** [comment]
Severely outdated

File: chat-plugins/info.js [hunk 1]
```diff
@@ -910,12 +910,11 @@ exports.commands = {
 	groups: function (target, room, user) {
 		if (!this.runBroadcast()) return;
 		this.sendReplyBox(
-			"+ <b>Voice</b> - They can use ! commands like !groups, and talk during moderated chat<br />" +
-			"% <b>Driver</b> - The above, and they can mute. Global % can also lock users and check for alts<br />" +
-			"@ <b>Moderator</b> - The above, and they can ban users<br />" +
-			"&amp; <b>Leader</b> - The above, and they can promote to moderator and force ties<br />" +
-			"# <b>Room Owner</b> - They are leaders of the room and can almost totally control it<br />" +
-			"~ <b>Administrator</b> - They can do anything, like change what this message says"
+			Config.groupsranking
```
Comment (AnnikaCodes) [inline chat-plugins/info.js:913]: This is a cool idea tbh; it's a shame the PR is abandoned.

**Comment by AnnikaCodes** [comment]
inactive for 3+ years, going to assume this is abandoned

**Pull request closed without merging**